include/seqan3/utility/type_list/traits.hpp

```diff
@@ -104,6 +104,40 @@ template <typename replace_t,
           typename ...pack_t>
 pack_traits::replace_at<replace_t, idx, pack_t...> replace_at(type_list<pack_t...>);
 
+//!\brief A helper for seqan3::list_traits::detail::unique [recursion anchor]
+template <typename query_t>
+type_list<> filter_out(type_list<>);
+
+//!\brief A helper for seqan3::list_traits::detail::unique [recursion]
+template <typename query_t, typename head_t, typename ...pack_t>
+auto filter_out(type_list<head_t, pack_t...>)
+{
+    if constexpr(std::same_as<query_t, head_t>)
+        return filter_out<query_t>(type_list<pack_t...>{});
+    else
+        return concat(type_list<head_t>{}, filter_out<query_t>(type_list<pack_t...>{}));
+}
```

Comment:
Maybe `remove` or `remove_all` would be a better name?

"remove `query_t` from the list `pack_t...`"
"remove all `query_t` from the list `pack_t...`"

`remove` (sometimes) implies that it stops after one hit, that's why `remove_all` might be better

"filter out" sounds a bit weird and the cpp20-me expects to pass a predicate :)

Reply:
We currently have
* `seqan3::list_traits::count<int, list_t>` (element) https://en.cppreference.com/w/cpp/algorithm/count
* `seqan3::list_traits::find<double, list_t>` (element) https://en.cppreference.com/w/cpp/algorithm/find
* `seqan3::list_traits::find_if<std::is_pointer, list_t>` (predicate) https://en.cppreference.com/w/cpp/algorithm/find
* `seqan3::list_traits::remove<double, list_t>` (element) https://en.cppreference.com/w/cpp/algorithm/remove

I like that suggestion, I'll rename it `remove`

Reply:
Because of your ingenious idea of using count == 1, I could throw away the unique and remove :)